File: DziennikElektroniczny/Models/Attendance.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data;
using System;

namespace DziennikElektroniczny.Models
{
    public class Attendance
    {
        [Key]
        public int AttendanceId { get; set; }

        [Range(1, 1000000000)]
        public int? StudentPersonId { get; set; }
        public Person StudentPerson { get; set; }

        [Range(1, 1000000000)]
        public int? LessonId { get; set; }
        public Lesson Lesson { get; set; }

        [Required]
        [Range(1, 1000000000)]
        public int WasPresent { get; set; }

        [MaxLength(2000)]
        public string AbsenceNote { get; set;}

    }
}

File: DziennikElektroniczny/AngularApp/frontend/src/app/models/Attendance.ts
export class AttendanceViewModel {
    constructor(
        public id: number = -1,
        public studentPersonId: number = -1,
        public studentDisplayName: string = '',
        public lessonId: number = -1,
        public subjectName: string = '',
        public lessonDate: string = '',
        public wasPresent: boolean = false,
        public absenceNote: string = '',
    ) {}
}
